src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js

```diff
@@ -441,52 +464,57 @@ const AnalysisEditorFrame = ({
   console.assert(!labels.welderInstallFailed, 'Expected cloud environment to have Welder')
   const frameRef = useRef()
   const [busy, setBusy] = useState(false)
-  const [notebookSetupComplete, setNotebookSetupComplete] = useState(false)
+  const [analysisSetupComplete, setAnalysisSetupComplete] = useState(false)
   const cookieReady = Utils.useStore(cookieReadyStore)
 
-  const localBaseDirectory = `${name}/edit`
-  const localSafeModeBaseDirectory = `${name}/safe`
-  //TODO: should this vary on toolLabel? (As of right now, the answer is no because the VM path will be `/home/rstudio`, but the bucket path for all analyses will be `/notebooks`, but design still WIP)
+  const localBaseDirectory = Utils.switchCase(toolLabel,
+    [tools.Jupyter.label, () => `${name}/edit`],
+    [toolLabel === tools.RStudio.label, () => ''])
+
+  const localSafeModeBaseDirectory = Utils.switchCase(toolLabel,
+    [tools.Jupyter.label, () => `${name}/safe`],
+    [tools.RStudio.label, () => ''])
+
   const cloudStorageDirectory = `gs://${bucketName}/notebooks`
 
-  //TODO: see inline TODOs, may require call-site change too
-  const setUpNotebook = _.flow(
+  const pattern = Utils.switchCase(toolLabel,
+    [tools.RStudio.label, () => `.*\\.Rmd`],
+    [tools.Jupyter.label, () => `.*\\.ipynb`])
```

Comment:
We try to keep any close braces on the same level of indentation as their matching open braces, so in this case they should be on new lines.
```suggestion
    [tools.Jupyter.label, () => `.*\\.ipynb`]
  )
```